### DISCUSSION TITLE
Encryption at rest

### DISCUSSION THREAD

**Discussion by christian-valadez** [supabase/supabase#3656] [Questions]
Are there any recipes for encrypting a table at rest? One of the benefits of Firebase Firestore is that [data is encrypted](https://cloud.google.com/firestore/docs/server-side-encryption) and I was wondering if there was a way to do this with Supabase / Postgres. I'm a bit of a newbie with Postgres, so any help with this would be greatly appreciated!

**Comment by awalias** [comment]
all the data you store is automatically encrypted at rest _at the file system level_ (presumably similar to how firebase is doing it where the keys are managed by google/supabase)

additionally, like Firebase, you have the option of doing client side encryption/decryption where you (or the user) manages the keys yourself/themselves. But this is application specific, you would typically use something like [AES](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt)
